src/data/model.ts

```diff
@@ -0,0 +1,35 @@
+export default class Model {
+  private translateFields = {};
+
+  public factory(data: any): any {
+    for (let key in data) {
+      if (data.hasOwnProperty(key)) {
+        let rawData = data[key];
+        key = this.translate(key);
+        if (!this.hasOwnProperty(key) || typeof rawData === "undefined") {
+          continue;
```

Comment:
I am not a big fan of this continue instead use the if statement to return true and put the below code inside if